llvm/test/CodeGen/SPIRV/transcoding/sub_group_ballot.ll

```diff
@@ -226,7 +226,7 @@ target triple = "spirv64-unknown-unknown"
 ; CHECK-SPIRV-DAG: %[[char_0:[0-9]+]] = OpConstant %[[char]] 0
 ; CHECK-SPIRV-DAG: %[[short_0:[0-9]+]] = OpConstant %[[short]] 0
 ; CHECK-SPIRV-DAG: %[[int_0:[0-9]+]] = OpConstant %[[int]] 0
-; CHECK-SPIRV-DAG: %[[long_0:[0-9]+]] = OpConstant %[[long]] 0
+; CHECK-SPIRV-DAG: %[[long_0:[0-9]+]] = OpConstantNull %[[long]]
```

Comment:
Now it seems a bit out-of-sync with the other constants around this test as you can see, why generate `OpConstantNull` for longs specifically?

Reply:
No special rule for longs, they are now processed by quite common GR.getOrCreateConstInt() in SPIRVInstructionSelector::selectConst, which constructs `OpConstantNull `for zero constants. It looks like all `Imm.getBitWidth() <= 32` constants are processed specially PseudoConstI in SPIRVInstrInfo.td and it can only generate `OpConstant`.

It's not a problem from the specification point of view. So I think we'll fix it later.